### ISSUE TITLE
SingleButton86 & DuplexButton86

### ISSUE THREAD

**Issue by 350d** [YinHangCode/homebridge-mi-aqara#142]
Hello! Thank you for your awesome plugin! Today I've tried to setup my first wireless single and double switches and can't get the point how to use it with HomeKit in proper way.
I have this in my `config.json`:
````
                "158d00015f2663": {
                    "SingleButton86_StatelessProgrammableSwitch": {
                        "name": "SB#1",
                        "syncValue": true
                    },
                    "SingleButton86_Switch_VirtualSinglePress": {
                        "name": "SB#1V",
                        "syncValue": true
                    }
                }
````
Questions:
1. `SingleButton86_StatelessProgrammableSwitch ` and `SingleButton86_Switch_VirtualSinglePress ` should be equal to title from Mi Home App or its doesn't matter?
2. `VirtualSinglePress` available in HomeKit as a Switch, but I can't turn it on, because its just flashes one on ON position and then turns off immediately. Why? And how I can fix it?

Also, I have this `WARN` in my homebridge log:
````
[WARN][Revc]{"cmd":"write_ack","model":"86sw1","sid":"158d00015f2663","short_id":8779,"data":"{\"voltage\":3055}"}
````

**Comment by klimf** [comment]
Have the same issue

**Comment by theo-69** [comment]
The Reason is, right now its designed to toggle the button if you press the switch. Reason for it, you can toggle the function from the miio app within the gateway with it.

**Comment by 350d** [comment]
@theo-69 Toggle doesn't work and this issue is about it.

**Comment by theo-69** [comment]
Well thats the thing, watch the associated button (with in home app), if you use the virtual switch the button will jump. It will than send the press to you gateway and run the scene you associated to the button within your gateway.

**Comment by 350d** [comment]
@theo-69 Yes, they are the physical buttons - thats why we have virtual switches additionally to buttons, to have an option to toggle this switch with single button click.

**Comment by theo-69** [comment]
That how we both would like to have it, however the author of this plugins intentions is the other way around not toggle switch with button. He does a virtuell button press with a switch to than send the information to the gateway. The gateway now thinks you pressed the button!

**Comment by 350d** [comment]
@theo-69 How did you known for sure about author logic? Can you check the sources? I think this is just a bug.

**Comment by theo-69** [comment]
Yep found that somewhere and if you look at the parser its intended to do so. That somehow even makes sense but confused me as well. Maybe we can find a way to change this with a Fork, tried already but failed. You will need to change at least four parser files SwitchVirtualBasePressParser.js and SingleButton86Parser.js, DuplexButton86Parser.js, ButtonParser.js

**Comment by 350d** [comment]
@theo-69 Now I get the point, maybe the ids behind it was about to make some automations with virtual switch, because automations not available for programmable switches (like these buttons). Anyway, it will be awesome to have a settings in config to tweak this functionality (toggle/click)

**Comment by theo-69** [comment]
yep, but I tried but just a newbie and lost with all those codes. Also this [fork](https://github.com/acc-ua/homebridge-mi-aqara-2) is interesting. However always the same, just added Sound playing Switches. There is a new attempt by using miio [here](https://github.com/pieter/homebridge-miio-gateway)!

**Comment by YinHangCode** [comment]
open VirtualSinglePress switch equivalent to press the physical button.
It will automatically jump back to close, because of he is a button, no state.

**Comment by 350d** [comment]
@YinHangCode is there any option to make it work like a switch to toggle on/off state by pressing single button?

**Comment by YinHangCode** [comment]
It should not be possible, because the button itself has no state.
In fact, HomeKit StatelessProgrammableSwitch Accessory do not support click operation, so a VirtualSinglePress is added.

**Comment by 350d** [comment]
@YinHangCode How Wemo Light Switches works in this case? They have `on` and `off` state and just controlled with a single button press? It should be possible to emulate this type of switches for sure, isn't it?

**Comment by theo-69** [comment]
tried it with no success, but it is possible because its used in the plugin for the amazon button.
